src/triage/component/postmodeling/plots.py

```diff
@@ -0,0 +1,211 @@
+        transparent = kwargs.get('transparent', False)
+        dpi = kwargs.get('dpi', 80)
+        bbox_inches = kwargs.get('bbox_inches', 'tight')
+        fig.savefig(f"{filename}.{extension}", transparent=transparent, dpi=dpi, bbox_inches=bbox_inches)
+
+def plot_roc(model, **kwargs):
+    fig, ax = plt.subplots()
+
+    predictions = get_predictions(model)
+    labels = predictions.label_value
+    scores = predictions.score
+    fpr, tpr, thresholds = roc_curve(labels, scores)
+    roc_auc = auc(fpr, tpr)
+
+    fmt = kwargs.get('fmt', 'b')
+
+    # Random performance
+    ax.plot([0, 1], [0, 1], 'k--')
+
+    # Actual model
+    ax.plot(fpr, tpr, fmt, label='ROC curve (area = %0.2f)' % roc_auc)
+
+    ax.set_xlim([0.0, 1.0])
+    ax.set_ylim([0.0, 1.0])
+
+    ax.set_xlabel('False Positive Rate')
+    ax.set_ylabel('True Positive Rate')
+
+    ax.set_title(kwargs.get('title', f"(Model {model.id} @ {model.train_end_time.strftime('%Y-%m-%d')})"))
+
+    ax.legend(loc="lower right")
+
+    fig.suptitle(kwargs.get('suptitle', f"Model Group {model.model_group.id}"))
+
+    _store(fig, **kwargs)
+
+    return fig, ax
+
+
+def plot_precision_recall_n(model, **kwargs):
+    fig, ax1 = plt.subplots()
+
+    predictions = get_predictions(model)
+    labels = predictions.label_value
+    scores = predictions.score
+
+    y_axis = np.arange(0, 101, 1)
+
+    prec_k = precision_n(labels, scores)
+    rec_k = recall_n(labels, scores)
+
+    prec_color = kwargs.get('prec_fmt', 'tab:blue')
+    rec_color = kwargs.get('rec_fmt', 'tab:red')
+
+    ax1.plot(y_axis, prec_k, prec_color)
+    ax1.set_xlabel('Percent of population')
+    ax1.set_ylabel('Precision', color=prec_color)
+    ax1.tick_params(axis='y', labelcolor=prec_color)
+
+    ax1.set_ylim([0,1])
+    ax1.set_xlim([0,100])
+
+    ax1.set_title(kwargs.get('title', f"(Model {model.id} @ {model.train_end_time.strftime('%Y-%m-%d')})"))
+
+    ax2 = ax1.twinx()
+    ax2.plot(y_axis, rec_k, rec_color)
+    ax2.set_ylim([0,1])
+    ax2.set_ylabel('Recall', color=rec_color)
+    ax2.tick_params(axis='y', labelcolor=rec_color)
+
+
+    fig.suptitle(kwargs.get('suptitle', f"Model Group {model.model_group.id}"))
+
+    _store(fig, **kwargs)
+
+    return fig, ax1, ax2
+
+def plot_metric_over_time(model_groups, metric, parameter, **kwargs):
+
+    def get_model_group_evaluations(model_group):
+        evaluations = pd.concat([get_evaluations(model).query(f"metric == '{metric}@' and parameter == '{parameter}'") for model in model_group])
+        evaluations = evaluations.sort_values('evaluation_start_time', ascending=True)
+        evaluations = evaluations[['evaluation_start_time', 'value']]
+        evaluations['type'] = model_group.type.split('.')[-1]
+        return evaluations
+
+    metric = metric.replace('@','')
+
+    fig, ax = plt.subplots()
+
+    evaluations = pd.concat([get_model_group_evaluations(model_group) for model_group in model_groups])
```

Comment:
superfluous list construction (and memory usage to boot)

```suggestion
    evaluations = pd.concat(get_model_group_evaluations(model_group) for model_group in model_groups)
```

Reply:
Nice...